jobbergateapi2/tests/apps/job_scripts/test_routers.py

```diff
@@ -442,3 +448,115 @@ async def test_list_job_script_all(client, user_data, application_data, job_scri
+    assert job_script.job_script_name == "new name"
+    assert job_script.updated_at == now
+
+
+@pytest.mark.asyncio
+@database.transaction(force_rollback=True)
+async def test_update_job_script_not_found(client, user_data, application_data, job_script_data):
+    """
+    Test update a job_script not found, must return 404 and do nothing with the stored data.
+    """
+    user = [UserCreate(id=1, **user_data)]
+    await insert_objects(user, users_table)
+
+    application = [Application(id=1, application_owner_id=1, **application_data)]
+    await insert_objects(application, applications_table)
+
+    job_scripts = [JobScript(id=1, **job_script_data)]
+    await insert_objects(job_scripts, job_scripts_table)
+
+    response = client.put("/job-scripts/123", data={"job_script_name": "new name"})
+
+    assert response.status_code == status.HTTP_404_NOT_FOUND
+
+    query = job_scripts_table.select(job_scripts_table.c.id == 1)
+    job_script = JobScript.parse_obj(await database.fetch_one(query))
+
+    assert job_script is not None
+    assert job_script.job_script_name == job_script_data["job_script_name"]
+
+
+@pytest.mark.asyncio
+@database.transaction(force_rollback=True)
+async def test_delete_job_script(client, user_data, application_data, job_script_data):
+    """
+    Test delete a job_script, must return 204 and delete the data.
+    """
+    user = [UserCreate(id=1, **user_data)]
+    await insert_objects(user, users_table)
+
+    application = [Application(id=1, application_owner_id=1, **application_data)]
+    await insert_objects(application, applications_table)
+
+    job_scripts = [JobScript(id=1, **job_script_data)]
+    await insert_objects(job_scripts, job_scripts_table)
+
+    count = await database.fetch_all("SELECT COUNT(*) FROM job_scripts")
+    assert count[0][0] == 1
+
+    response = client.delete("/job-scripts/1")
+
+    assert response.status_code == status.HTTP_204_NO_CONTENT
+
+    count = await database.fetch_all("SELECT COUNT(*) FROM job_scripts")
+    assert count[0][0] == 0
+
+
+@pytest.mark.asyncio
+@database.transaction(force_rollback=True)
+async def test_delete_job_script_not_found(client, user_data, application_data, job_script_data):
+    """
+    Test delete a job_script not found, must return 404 and do nothing with the data.
+    """
+    user = [UserCreate(id=1, **user_data)]
+    await insert_objects(user, users_table)
+
+    application = [Application(id=1, application_owner_id=1, **application_data)]
+    await insert_objects(application, applications_table)
+
+    job_scripts = [JobScript(id=1, **job_script_data)]
+    await insert_objects(job_scripts, job_scripts_table)
+
+    count = await database.fetch_all("SELECT COUNT(*) FROM job_scripts")
+    assert count[0][0] == 1
+
+    response = client.delete("/job-scripts/123")
+
+    assert response.status_code == status.HTTP_404_NOT_FOUND
+
+    count = await database.fetch_all("SELECT COUNT(*) FROM job_scripts")
+    assert count[0][0] == 1
```

Comment:
@jamesbeedy here is another good use of the count, since it is way easier and faster to just verify if the actions affected the database